lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoFileSystem.java

```diff
@@ -16,30 +16,138 @@
+     * will never be checked. The file location path cannot be empty, and must not end with a slash
+     * or whitespace.
+     *
+     * @throws IllegalArgumentException if location is not valid for this file system
+     */
     TrinoInputFile newInputFile(String location, long length);
 
+    /**
+     * Creates a TrinoOutputFile which can be used to create or overwrite the file. The file
+     * location path cannot be empty, and must not end with a slash or whitespace.
+     *
+     * @throws IllegalArgumentException if location is not valid for this file system
+     */
     TrinoOutputFile newOutputFile(String location);
 
+    /**
+     * Deletes the specified file. The file location path cannot be empty, and must not end with
+     * a slash or whitespace.
+     *
+     * @throws IllegalArgumentException if location is not valid for this file system
+     * @throws IOException if the file does not exist or was not deleted.
+     */
     void deleteFile(String location)
             throws IOException;
 
     /**
-     * Delete files in batches, possibly non-atomically.
-     * If an error occurs, some files may have been deleted.
+     * Delete specified files. This operation is <b>not</b> required to be atomic, so if an error
+     * occurs, all, some, or, none files may be deleted. This operation may be faster than simply
+     * looping over the location as some file systems support batch delete operations natively.
+     *
+     * @throws IllegalArgumentException if location is not valid for this file system
+     * @throws IOException if a file does not exist or was not deleted.
      */
-    void deleteFiles(Collection<String> locations)
-            throws IOException;
+    default void deleteFiles(Collection<String> locations)
+            throws IOException
+    {
+        for (String location : locations) {
+            deleteFile(location);
+        }
+    }
 
+    /**
+     * Deletes all files and directories within the specified directory recursively, and deletes
+     * the directory itself. If the location does not exist, this method is a noop. If the location
+     * does not have a path, all files and directories in the file system are deleted.
+     * <p>
+     * For hierarchical file systems (e.g. HDFS), if the path is not a directory, an exception is
+     * raised.
+     * <p>
+     * For blob file systems (e.g., S3), if the location does not end with a slash one is appended,
+     * and all blobs that start with that prefix are deleted.
+     * <p>
+     * If this operation fails, some, none, or all of the contents of the directory may
+     * have been deleted.
+     *
+     * @param location the directory to delete
+     * @throws IllegalArgumentException if location is not valid for this file system
+     */
     void deleteDirectory(String location)
             throws IOException;
 
+    /**
+     * Rename source to target without overwriting target.  This method is not required
```

Comment:
The only usage today is in `AzureTransactionLogSynchronizer` which expects the method to be atomic. I'm wondering if we should require it to be atomic, maybe with a way to check if atomic rename is supported?